contracts/contracts/strategies/ConvexEthMetaStrategy.sol

```diff
@@ -257,19 +337,169 @@ contract ConvexEthMetaStrategy is InitializableAbstractStrategy {
+            address(this)
+        );
+
+        // The vault burns the OTokens from this strategy
+        IVault(vaultAddress).burnForStrategy(oethToBurn);
+
+        emit Withdrawal(address(oeth), address(lpToken), oethToBurn);
+    }
 
-        // Send all ETH and WETH on the contract, including extra
-        weth.deposit{ value: address(this).balance }();
+    /**
+     * @notice Mint OTokens and one-sided add to the Metapool.
+     * This is used when the Metapool has not enough OTokens and too many ETH.
+     * The OToken/Asset, eg OETH/ETH, price with increase.
+     * The amount of assets in the vault is unchanged.
+     * The total supply of OTokens is increased.
+     * The asset value of the strategy and vault is increased.
+     * @param _oTokens The amount of OTokens to be minted and added to the pool.
+     */
+    function mintAndAddOTokens(uint256 _oTokens)
+        external
+        onlyStrategist
+        improveMetapoolBalance
+    {
+        IVault(vaultAddress).mintForStrategy(_oTokens);
+
+        // Convert OETH to Metapool LP tokens
+        uint256 valueInLpTokens = (_oTokens).divPrecisely(
+            curvePool.get_virtual_price()
+        );
+        // Apply slippage to LP tokens
+        uint256 minMintAmount = valueInLpTokens.mulTruncate(
+            uint256(1e18) - MAX_SLIPPAGE
+        );
+
+        // Add the minted OTokens to the Curve Metapool
+        uint256 lpDeposited = curvePool.add_liquidity(
+            [0, _oTokens],
+            minMintAmount
+        );
+
+        // Deposit the Metapool LP tokens to the Convex rewards pool
+        require(
+            IConvexDeposits(cvxDepositorAddress).deposit(
+                cvxDepositorPTokenId,
+                lpDeposited,
+                true // Deposit with staking
+            ),
+            "Failed to Deposit LP to Convex"
+        );
+
+        emit Deposit(address(oeth), address(lpToken), _oTokens);
+    }
+
+    /**
+     * @notice One-sided remove of ETH from the Metapool, convert to WETH
+     * and transfer to the vault.
+     * This is used when the Metapool has not enough OTokens and too many ETH.
+     * The OToken/Asset, eg OETH/ETH, price with decrease.
+     * The amount of assets in the vault increases.
+     * The total supply of OTokens does not change.
+     * The asset value of the strategy reduces.
+     * The asset value of the vault should be close to the same.
+     * @param _lpTokens The amount of Metapool LP tokens to be burned for ETH.
+     * @dev Metapool LP tokens is used rather than WETH assets as Curve does not
+     * have a way to accurately calculate the amount of LP tokens for a required
+     * amount of ETH. Curve's `calc_token_amount` functioun does not include fees.
+     * A 3rd party libary can be used that takes into account the fees, but this
+     * is a gas intensive process. It's easier for the trusted strategist to
+     * caclulate the amount of Metapool LP tokens required off-chain.
```

Comment:
agree, though IMO the strategist should also pass the other side of the trade - min amount of eth expected